### PR TITLE
[hw,spi_device,dv] Enable CDC instrumentation

### PR DESCRIPTION
**Pull request by crteja**
- Enable CDC randomisation
- Update FLASH_STATUS register update timing to account for CDC randomisation enabled with this change
This PR partially addresses https://github.com/lowRISC/opentitan/issues/16689

### CODE REVIEW

**Review by andreaskurth — APPROVED** [review]
LGTM, thanks @crteja

**Review by hcallahan-lowrisc — APPROVED** [review]

**Review by marnovandermaas — APPROVED** [review]
LGTM, let's wait for CI to pass before merging.

**Pull request merged**
